firefox_ui_tests/functional/locationbar/test_star_in_autocomplete.py

```diff
@@ -66,11 +73,9 @@ def test_star_in_autocomplete(self):
         self.wait_for_condition(lambda mn: len(autocomplete_results.visible_results) == 1)
 
         first_result = autocomplete_results.visible_results[0]
-        matching_title = autocomplete_results.get_matching_text(first_result, 'title')[0]
-        self.assertEqual(
-            search_string.lower(),
-            matching_title.lower(),
-            'The page title matches the highlighted text')
-        self.assertEqual('bookmark',
-                         first_result.get_attribute('type'),
-                         'The auto-complete result is a bookmark')
+        matching_titles = autocomplete_results.get_matching_text(first_result, 'title')
+        for title in matching_titles:
+            self.wait_for_condition(lambda mn: search_string.lower() == title.lower())
```

Comment:
Please put the expected value as second inside a comparison, that makes it easier to see what we actually want to test. Also search_string is already lower-cased by definition.